### PR TITLE
AssertThrow for BlockVector usage in MassOperator

### PR DESCRIPTION
**Pull request by gfcas**
1. Maybe highlight this note in the documentation of the MassOperator a bit more. I just run intho this trap.
2. Add an AssertThrow in the constructor of this class to catch the not implemented case.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: include/deal.II/matrix_free/operators.h [hunk 1]
```diff
@@ -1831,7 +1831,12 @@ namespace MatrixFreeOperators
                VectorType,
                VectorizedArrayType>::MassOperator()
     : Base<dim, VectorType, VectorizedArrayType>()
-  {}
+  {
+    AssertThrow(
+      IsBlockVector<VectorType>::value == false,
+      ExcNotImplemented(
+        "This class only supports the non-blocked vector variant of the Base operator because only a single FEEvaluation object is used in the apply function."));
```
Comment (kronbichler) [inline include/deal.II/matrix_free/operators.h:1838]: Can you break the line before 80 characters? Looks good otherwise.

**Comment by gfcas** [comment]
Just a final question: Is a AssertThrow OK or better just a Assert, which is only triggered in the DEBUG mode?

**Review by kronbichler — APPROVED** [review]
Let's go with this variant.

**Pull request merged**